src/engine/Distinct.cpp

```diff
@@ -17,7 +17,15 @@ size_t Distinct::getResultWidth() const { return subtree_->getResultWidth(); }
 Distinct::Distinct(QueryExecutionContext* qec,
                    std::shared_ptr<QueryExecutionTree> subtree,
                    const std::vector<ColumnIndex>& keepIndices)
-    : Operation{qec}, subtree_{std::move(subtree)}, keepIndices_{keepIndices} {}
+    : Operation{qec}, subtree_{std::move(subtree)}, keepIndices_{keepIndices} {
+  AD_CORRECTNESS_CHECK(subtree_);
+  AD_CORRECTNESS_CHECK(ql::ranges::all_of(
+      keepIndices_, [sortedCols = subtree_->resultSortedOn(),
+                     size = keepIndices_.size()](ColumnIndex distinctCol) {
+        return ad_utility::contains(std::span{sortedCols}.subspan(0, size),
+                                    distinctCol);
+      }));
+}
```

Comment:
For consistency, I would prefer the `createSortedTree` function here, and passing in the subtrees unsorted.
If you want to, you can implement the version of `createSortedTree` that also accepts an arbitrary permutation of the sort columns.
The you can revert all the changes in tests etc. and also remove some explicit calls to the sorting in the `QueryPlanner.cpp`.